File: javascript/firstOccurance.js
function findFirstOccurance(nums, target){
    let left = 0;
    let right = nums.length-1;
    
    while(left <= right){
        let mid = Math.floor(left + (right - left)/2);

        if(nums[mid] === target && (mid === 0 || nums[mid-1] !== target)){
            return mid;
        }else if(nums[mid] < target){
            left = mid + 1;
        }else{
            right = mid - 1;
        }
    }

    return -1;
}

console.log(findFirstOccurance([2, 4, 6, 6, 6, 8, 10], 6));

File: Java/FirstOccurance.java
public class FirstOccurance {
    private static int findFirstOccurance(int[] nums, int target, int i){
        if(nums.length == i) return -1;

        if(nums[i] == target){
            return i;
        }

        return findFirstOccurance(nums, target, i+1);
    }
    public static void main(String[] args) {
        int[] nums = {8, 3, 6, 9, 5, 3, 2, 5, 9};
        System.out.println(findFirstOccurance(nums, 5, 0));
    }
}
